### PR TITLE
OpenSSF Best Practices Program Original Article

### LINKED ISSUE #807: OpenSSF Best Practices Badge Program
Author: @bartlettroscoe 
EB member: @rinkug 

https://bestpractices.coreinfrastructure.org/en is sponsored by the Linux Foundation.  It is a badging program for software best practices.  There are multiple levels, including "passing", silver, and gold.  Criteria are fairly extensive.

I haven't reviewed this thoroughly yet for applicability to scientific software.  I think it is more aimed at system software and such.  But I suspect most or all could be applicable in CSE at some level.  This might be something to discuss a bit in a CC writeup.

### PR DESCRIPTION
**Pull request by bartlettroscoe**
# Description

EB Member: @pagrubel

Resolves #807

This turned out to be a bit of a longer article.  But I am considering this an original article.  It is about the length of a short conference article (about 8 pages when printed).  But there are several figures, tables, and lists so the text is not very dense.

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [x] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* [x] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).
* [x] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [x] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [x] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [x] ***[EB Mem]*** Add one or more Reviewers.
* [x] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [x] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [x] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [x] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [x] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [x] ***[Author]*** Make any final changes to the PR based on feedback.
* [x] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).
* [x] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [x] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [x] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [x] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [ ] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
So is rebuilt the preview.bssw.io site and it produced:

* https://preview.bssw.io/items/the-openssf-best-practices-badge-program

I think I should take "The" out of the title to give this a better name and "The" is not needed.

So the `<sup>...</sup>` seem to be displayed correctly everywhere on bssw.io where they are not in my local Chrome Extension Markdown viewer or on the GitHub markdown viewer.

For example the line:

```
The OpenSSF Best Practices are listed for each badge level individually and as a full set.<sup>[ossfbpa]</sup>
```

on GitHub [here](https://github.com/bartlettroscoe/bssw.io/blob/807-openssf-best-practices/Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md) displays "ossfbpa" as not superscripted as:

image

but on bssw.io, it displays correctly as a superscript as:

image

So this is one case where the bssw.io site generator correctly interprets the Markdown where GitHub (an the Chrome Extension) does not.  Therefore, I think the rendering on bssw.io is okay.

But I am seeing that I need to change link text for the references like ""[description_good]" or just remove them.

**Comment by bartlettroscoe** [comment]
CC: @bernhold, @rinkug 

@pagrubel, I think this original article is ready to review if you have time.  I had my wife read it for English issues so hopefully there are not too many of those mistakes left.

I am not to sure about a few things:

* Should the article keep the longer acronyms for the references like "ossfbpbp" or just replace them with numbers using wikize-refs.py?
* What reference titles should be used for the best practices?  (Currently, they just repeat the short identifier like "description_good")

There are a few things that I am seeing now that definitely need to be fixed:

* The columns from the table "Table: OpenSSF Best Practice Breakdown" should be removed that don't fit in the bssw.io rendering?  (I think the obvious answer is yes and we should just link to the full table on the best practices site)
* The figures are not displayed correctly.  (How do we blow them up to take the width of the screen?)
* I think all of the bullets in the introduction should link to material below, to act as a type of table of contents for the short document.

**Comment by bartlettroscoe** [comment]
Okay, I think I have finally gotten the images to display as they should in the last rendering:

* https://preview.bssw.io/items/openssf-best-practices-badge-program

I don't fully understand why they did not display correctly with commit https://github.com/betterscientificsoftware/bssw.io/pull/1367/commits/b3d4d14d670fee7bcfeda0bb4d9df65588cb179a and then why I had to remove the image captions.

It took over 2 hours with many experimentation iterations (each taking over 10 minutes waiting for the preview.bssw.io site to rebuild) before I got the images to display correctly.  (But I think some of my iterations should have displayed the images correctly but instead just give non-rendered icons.  Something very strange is going on with the rebuild process that seemed to defy reproduction.)

But the images are being rendered in very large vertical margins showing, for example:

image

Why would the bssw.io site generator do that?

**Comment by markcmiller86** [comment]
> I don't fully understand why they did not display correctly with commit [b3d4d14](https://github.com/betterscientificsoftware/bssw.io/commit/b3d4d14d670fee7bcfeda0bb4d9df65588cb179a) and then why I had to remove the image captions.

I dunno if it helps but I looked for other instances of the use of `lightbox` styling of images and see this one (around line 45)...

https://raw.githubusercontent.com/betterscientificsoftware/bssw.io/c3819bb1934da4555a26544c50e717050ca5d851/Articles/Blog/PortingToHIP.md

And see a difference between your typsetting of that and the one used in the above article.

Do you need `lightbox` styling? I am not entirely sure. If you don't, doesn't generic markdown image treatmeant work...

``

**Comment by bartlettroscoe** [comment]
> Do you need `lightbox` styling? I am not entirely sure. If you don't, doesn't generic markdown image treatment work...
> 
> ``

@markcmiller86, according the the [official bssw.io image documentation](https://github.com/betterscientificsoftware/bssw.io/blob/master/images/README.md#how-to-place-an-image-on-a-page), the only way to scale an image up to full page width is to use `class='page lightbox`.

Looking at the rendered PortToHIP.md article at:

* https://bssw.io/blog_posts/porting-the-ginkgo-package-to-amd-s-hip-ecosystem

it also shows excessive vertical spacing for one of the figures:

image

On a side note, what is interesting about:

* https://raw.githubusercontent.com/betterscientificsoftware/bssw.io/c3819bb1934da4555a26544c50e717050ca5d851/Articles/Blog/PortingToHIP.md

compared to the official documentation is that it puts `[caption]` on the left `[caption]` instead of the right`[caption]`.  Using `[caption]` does not appear to be standard HTML anyway according [here](https://www.w3schools.com/tags/tag_figcaption.asp) which says to use `<figcaption>`.

The process to experiment with formatting on bssw.io can take **several hours** if you need many experiments.

I just wish bssw.io  used a basic Jekyll site or something that I could build and test locally.  That would massively speed up this process.

**Comment by pagrubel** [comment]
Sorry, I'm just now getting to this, Ross. I've been out for a while. Great article! I do have a couple of minor English corrections that I'll suggest in a bit. I do think numbers would be better for the references than something as long as documentation_architecture. They just seem awkward to me.

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+#### Publication date: June 25, 2022
+ 
+The Linux Foundation's<sup>[lf]</sup> Open Source Security Foundation (OpenSSF)<sup>[openssf]</sup> is an effort to improve the quality and security of open source software.
+The OpenSSF Best Practices Badge Program<sup>[ossfbpbp],[ossfbpbgh]</sup> is an effort to collect, document, advocate, and help apply a comprehensive collection of best practices from the open source software development community to a wide rage of open source software projects.
+The foundation of this badge program is a "Badge App" site<sup>[ossfbpba]</sup> that allows projects to create an entry for their project and then to specify which practices are met, descriptions for how they meet the criteria for each practice, URLs to evidence, and justification for how each criteria is met (or why it is not met).
+
+As of 6/22/2022, the "Badge App" site has 4819 registered projects of which 789 have achieved a Passing-level badge.
+In addition, the community registers approximately 700 new open source projects on the site each year and approximately 158 projects achieve a Passing-level badge each year for the last several years.<sup>[ossfbpbas]</sup>
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:17]:
```suggestion
In addition, the community registers approximately 700 new open source projects on the site each year and approximately 158 projects have achieve a Passing-level badge each year for the last several years.<sup>[ossfbpbas]</sup>
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+These data suggest this program is a well accepted, well adopted badge program.
+
+The OpenSSF Best Practices and the supporting Badge App site enables a number of benefits by providing:
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:20]:
```suggestion
The OpenSSF Best Practices and the supporting Badge App site enable a number of benefits by providing:
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+
+### Badge App site
+
+One of the more unique features of the OpenSSF Best Practices program is the OpenSSF Best Practices "Badge App" site:
+
+* https://bestpractices.coreinfrastructure.org/en/projects
+
+This site allows project maintainers to create an entry for their project (which then creates a unique project ID) and go through the process of reading about each best practice item and marking each as "Met", "Unmet", "N/A", or "?".
+(Again, only a subset of practices allow an "N/A" response.)
+An example of an project's best practice item marked as "Met" with justification and a URL to evidence is given below:
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:131]:
```suggestion
An example of a project's best practice item marked as "Met" with justification and a URL to evidence is given below:
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+
+The OpenSSF Best Practices and the supporting Badge App site enables a number of benefits by providing:
+
+* a set of [best practices](#details) that have specific actionable criteria which require supporting evidence,
+
+* a featureful ["Badge App" site](#badge_app) that enhances the display of each practice, expanded descriptions of the practices, and fields to enter URL and text descriptions of the status of each practice in the project, 
+
+* a [badge](#badge_display) that can be displayed on a project's own hosting site to show that a project follows accepted best practices,
+
+* a particularly strong focus on [software security](#focus_on_security) which addresses the White House Executive Order 14028 "Improving the Nation's Cybersecurity"<sup>[eo14028]</sup> (which is of particular importance to U.S. laboratory and government agencies and contractors, including the U.S. national laboratories),
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:28]:
```suggestion
* a particularly strong focus on [software security](#focus_on_security) which addresses the White House Executive Order 14028 "Improving the Nation's Cybersecurity"<sup>[eo14028]</sup> (of particular importance to U.S. laboratory and government agencies and contractors, including the U.S. national laboratories),
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+  * Quality
+  * Security
+  * Analysis
+
+As noted above, some of the practices can be opted out as not applicable (i.e. "N/A") which is quite important in order for the OpenSSF Best Practices to be broadly applicable.
+For example, almost all of the technical security-related items can be marked as "N/A" if the project does not have any security-related or security-critical code.
+
+The OpenSSF Best Practices are listed for each badge level individually and as a full set.<sup>[ossfbpa]</sup>
+They are also listed with and without the detailed description/justification and links to more information.
+Each best practice is composed of a short statement of the practice (with the word MUST, SHOULD, or SUGGESTED) and a short HTML anchor/link name (which serves as a link to the practice and as an unambiguous shorthand identifier for that best practice item).
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:73]:
```suggestion
Each best practice is composed of a short statement of the practice (with the word MUST, SHOULD, or SUGGESTED) and a short HTML anchor/link name (that serves as a link to the practice and as an unambiguous shorthand identifier for that best practice item).
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+For example, of the over 900 pages in the book "Code Complete: 2nd Edition"<sup>[cc2nd]</sup>, exactly one paragraph is devoted to the area of software security in section 3.5 "Architecture Prerequisite":
+
+> The architecture should describe the approach to design level and code level security.
+> If a thread model has not previously been built, it should be built at architecture time.
+> Coding guidelines should be developed with security implications in mind, including approaches to handling buffers, rules for handling untrusted data (data input from users, cookies, configuration data, and other external interfaces), encryption, level of detail contained in error messages, protected secrete data that's in memory, and other issues.
+
+In addition, other standard texts on software engineering best practices never even mention the word "security".
+
+This increased focus on software security is consistent with White House Executive Order 14028 "Improving the Nation's Cybersecurity"<sup>[eo14028]</sup> which requires government and DOE software to improve software security.
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:216]:
```suggestion
This increased focus on software security is consistent with White House Executive Order 14028 "Improving the Nation's Cybersecurity"<sup>[eo14028]</sup> that requires government and DOE software to improve software security.
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+The foundation of this badge program is a "Badge App" site<sup>[ossfbpba]</sup> that allows projects to create an entry for their project and then to specify which practices are met, descriptions for how they meet the criteria for each practice, URLs to evidence, and justification for how each criteria is met (or why it is not met).
+
+As of 6/22/2022, the "Badge App" site has 4819 registered projects of which 789 have achieved a Passing-level badge.
+In addition, the community registers approximately 700 new open source projects on the site each year and approximately 158 projects achieve a Passing-level badge each year for the last several years.<sup>[ossfbpbas]</sup>
+These data suggest this program is a well accepted, well adopted badge program.
+
+The OpenSSF Best Practices and the supporting Badge App site enables a number of benefits by providing:
+
+* a set of [best practices](#details) that have specific actionable criteria which require supporting evidence,
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:22]:
```suggestion
* a set of [best practices](#details) that have specific actionable criteria that require supporting evidence,
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+
+<a name="badge_app"/>
+
+### Badge App site
+
+One of the more unique features of the OpenSSF Best Practices program is the OpenSSF Best Practices "Badge App" site:
+
+* https://bestpractices.coreinfrastructure.org/en/projects
+
+This site allows project maintainers to create an entry for their project (which then creates a unique project ID) and go through the process of reading about each best practice item and marking each as "Met", "Unmet", "N/A", or "?".
```
Comment (pagrubel) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:129]:
```suggestion
This site allows project maintainers to create an entry for their project (that then creates a unique project ID) and go through the process of reading about each best practice item and marking each as "Met", "Unmet", "N/A", or "?".
```

**Review by pagrubel — APPROVED** [review]

**Comment by pagrubel** [comment]
@bartlettroscoe I probably should have let you decide if you wanted to commit those changes.

**Comment by pagrubel** [comment]
@bartlettroscoe Do you think this is ready to publish. Would you want @karbarz to go over it?

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe I probably should have let you decide if you wanted to commit those changes.

@pagrubel, the commits you added look great.

> I do think numbers would be better for the references than something as long as documentation_architecture. They just seem awkward to me.

Okay, let me try that out and see how it looks.

**Comment by bartlettroscoe** [comment]
@karbarz, I think the only thing left to do is for you to look over this for issues before it is published.

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 9]
```diff
@@ -0,0 +1,414 @@
...
+
+* a particularly strong focus on [software security](#focus_on_security) which addresses the White House Executive Order 14028 "Improving the Nation's Cybersecurity"<sup>[8]</sup> (of particular importance to U.S. laboratory and government agencies and contractors, including the U.S. national laboratories),
+
+* a [learning tool](#learning_tool) for best practices for developers and projects,
+
+* a [roadmap for continual improvement](#continuous_improvement) for a project as it incrementally adopts more practices and improves its scores in different areas,
+
+* a [standard index](#standard_project_index) into the parts of the projects and how it handles different types of processes,
+
+* a [website template](#forking_badge_app_website) and database implementation that can be forked and customized for more targeted communities,
```
Comment (karbarz) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:36]:
```suggestion
* a [website template](#forking_badge_app_website) and database implementation that can be forked and customized for more targeted communities.
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 10]
```diff
@@ -0,0 +1,414 @@
...
+
+Note that some practices that are listed as SUGGESTED or SHOULD at a lower level are re-listed as MUST at a higher level.  For example, the ***Silver*** level lists the "bus factor" practice as SHOULD:
+
+* The project **SHOULD** have a "bus factor" of 2 or more. {Met URL} <sup>[bus_factor](https://bestpractices.coreinfrastructure.org/en/criteria#1.bus_factor)</sup>
+
+but at the ***Gold*** level, it relists it as MUST:
+
+* The project **MUST** have a "bus factor" of 2 or more. {Met URL} <sup>[bus_factor](https://bestpractices.coreinfrastructure.org/en/criteria#2.bus_factor)</sup>
+
+(That is why the number of "Total active" practices in the Silver and Gold levels are higher than the number "New at this  level".)
```
Comment (karbarz) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:118]:
```suggestion
(That is why the number of "Total active" practices in the Silver and Gold levels are higher than the number "New at this level".)
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 11]
```diff
@@ -0,0 +1,414 @@
...
+
+<a name="forking_badge_app_website"/>
+
+## Forking and Customizing the OpenSSF Best Practices and Badge App Website
+
+While the OpenSSF Best Practices are very comprehensive and are apparently well supported by the broader open source software development community, many projects are not fully open source and therefore some of the practices do not apply.
+For example, the MUST item \[floss_license\]<sup>[20]</sup> requires the software be released as Free-Libre /Open Source Software (FLOSS).
+That is very important for open source software but is not appropriate for many types of proprietary software.
+But most of the other OpenSSF Best Practices are applicable to all types of software.
+So how can other software development communities and organizations take advantage of the applicable OpenSSF Best Practices while excluding those that don't apply; and also add customized criteria for their community or organization?
```
Comment (karbarz) [inline Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:293]:
```suggestion
So, how can other software development communities and organizations take advantage of the applicable OpenSSF Best Practices while excluding those that don't apply, and also add customized criteria for their community or organization?
```

File: Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md [hunk 12]
```diff
@@ -0,0 +1,414 @@
...
+ 
+The Linux Foundation's<sup>[0]</sup> Open Source Security Foundation (OpenSSF)<sup>[1]</sup> is an effort to improve the quality and security of open source software.
+The OpenSSF Best Practices Badge Program<sup>[2],[3]</sup> is an effort to collect, document, advocate, and help apply a comprehensive collection of best practices from the open source software development community to a wide rage of open source software projects.
+The foundation of this badge program is a "Badge App" site<sup>[4]</sup> that allows projects to create an entry for their project and then to specify which practices are met, descriptions for how they meet the criteria for each practice, URLs to evidence, and justification for how each criteria is met (or why it is not met).
+
+As of 6/22/2022, the "Badge App" site has 4819 registered projects of which 789 have achieved a Passing-level badge.
+In addition, the community registers approximately 700 new open source projects on the site each year and approximately 158 projects have achieved a Passing-level badge each year for the last several years.<sup>[5]</sup>
+These data suggest this program is a well accepted, well adopted badge program.
```
[the remark this answers is not in the document]
Comment (bartlettroscoe) [reply Articles/ShortArticles/OpenssfBestPracticesBadgeProgram.md:18]: Interestingly, I originally had the hyphens "well-accepted" and "well-adopted" but my wife told me to remove them.  She said I had too many hyphens :-)

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe small suggestions for grammar changes

Thanks @karbarz, your suggestions look great (mostly adding comas).  I was able to commit them all in one batch commit https://github.com/betterscientificsoftware/bssw.io/pull/1367/commits/34309dccf83d4f4884b2aba821cc36c69ca16082(which is a new very welcome GitHub feature).

**Comment by bartlettroscoe** [comment]
@betterscientificsoftware/bssw-maint, both @pagrubel and @karbarz have reviewed and made several grammatical fixes.

This should be ready to merge and publish in this month's BSSw.io digest.

**Comment by bartlettroscoe** [comment]
> The subtitles in the article look too big formatwise on the preview website, right? I will try to figure that out.

@rinkug, yea, I agree.  That is strange because I used H2 ## for the subsections and H3 ### for the subsubsections so I don't know why they look as big as the title text which uses the main title H1 #.  I think this is an example of the problem that @bernhold raises in #1327.

**Comment by bernhold** [comment]
> > The subtitles in the article look too big formatwise on the preview website, right? I will try to figure that out.
> 
> @rinkug, yea, I agree. That is strange because I used H2 ## for the subsections and H3 ### for the subsubsections so I don't know why they look as big as the title text which uses the main title H1 #. I think this is an example of the problem that @bernhold raises in #1327.

I suggest using H3 for your first subdivision and bold text for the second.  That's what we do in the blogs.

**Comment by bartlettroscoe** [comment]
Just to follow up on the [above](https://github.com/betterscientificsoftware/bssw.io/pull/1367#issuecomment-1170412790) discussion of heading font sizes, changed the titles in commit https://github.com/betterscientificsoftware/bssw.io/pull/1367/commits/466d30866952355af4a1e80d90c6301b87774dee to use H3 ### for the section heads and bold `**<text>**` for subsections. You can see the result at:

* https://preview.bssw.io/items/openssf-best-practices-badge-program

The H1 # title font size looks like:

image

The H3 ### font size and the `**bolded**` subsection fort sizes looks like:

image

As you can see, there is a very step down in font size from the title font to the section headers.

I guess that looks okay.

**Comment by bartlettroscoe** [comment]
Item is published on main site:

* https://bssw.io/items/openssf-best-practices-badge-program

**Pull request merged**